pulp_ansible/tests/functional/api/test_rbac.py

```diff
@@ -0,0 +1,248 @@
+import pytest
+import uuid
+
+from pulp_ansible.tests.functional.constants import ANSIBLE_GALAXY_URL
+
+from pulp_ansible.tests.functional.utils import gen_ansible_client, gen_ansible_remote
+
+from pulp_smash.pulp3.bindings import monitor_task
+from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+
+from pulpcore.client.pulp_ansible import (
+    ApiException,
+    AsyncOperationResponse,
+    DistributionsAnsibleApi,
+    RepositoriesAnsibleApi,
+    RepositoriesAnsibleVersionsApi,
+    RemotesCollectionApi,
+    RemotesGitApi,
+    RemotesRoleApi,
+)
+
+
+def try_action(user, client, action, outcome, *args, **kwargs):
+    action_api = getattr(client, f"{action}_with_http_info")
+    try:
+        with user:
+            response, status, _ = action_api(*args, **kwargs, _return_http_data_only=False)
+        if isinstance(response, AsyncOperationResponse):
+            response = monitor_task(response.task)
+    except ApiException as e:
+        assert e.status == outcome, f"{e}"
+    else:
+        assert status == outcome, f"User performed {action} when they shouldn't been able to"
+        return response
+
+
+def test_ansible_repository_rbac(gen_user):
+    ansible_client = gen_ansible_client()
+    repo_api = RepositoriesAnsibleApi(ansible_client)
+
+    user_creator = gen_user(model_roles=["ansible.ansiblerepository_creator"])
+    user_reader = gen_user(model_roles=["ansible.ansiblerepository_viewer"])
+    user_helpless = gen_user()
+
+    # List testing
+    try_action(user_creator, repo_api, "list", 200)
+    try_action(user_reader, repo_api, "list", 200)
+    try_action(user_helpless, repo_api, "list", 200)
+
+    # Create testing
+    repo = try_action(user_creator, repo_api, "create", 201, gen_repo())
+    try_action(user_reader, repo_api, "create", 403, gen_repo())
+    try_action(user_helpless, repo_api, "create", 403, gen_repo())
+
+    # View testing
+    try_action(user_creator, repo_api, "read", 200, repo.pulp_href)
+    try_action(user_reader, repo_api, "read", 200, repo.pulp_href)
+    try_action(user_helpless, repo_api, "read", 200, repo.pulp_href)
```

Comment:
```suggestion
    try_action(user_helpless, repo_api, "read", 404, repo.pulp_href)
```
Queryset scoping prevents unauthorized users from even knowing the object exists.